src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets

```diff
@@ -151,14 +151,22 @@
       <_ToolchainPrefix Include="x86_64">
         <ToolPrefix>x86_64-linux-android</ToolPrefix>
       </_ToolchainPrefix>
+      <_ToolName Include="as"/>
+      <_ToolName Include="ld"/>
+      <_ToolName Include="strip"/>
     </ItemGroup>
 
+    <CreateItem
+	Include="@(_ToolchainPrefix)" AdditionalMetadata="Tool=%(_ToolName.Identity)">
+      <Output TaskParameter="Include" ItemName="_ToolchainTool" />
+    </CreateItem>
+
     <Which
         HostOS="$(HostOS)"
         HostOSName="$(HostOsName)"
-        Program="$(AndroidNdkFullPath)/toolchains/%(_ToolchainPrefix.Identity)-4.9/prebuilt/$(_PrebuiltToolchainDir)/bin/%(_ToolchainPrefix.ToolPrefix)-as"
+        Program="$(AndroidNdkFullPath)/toolchains/%(_ToolchainTool.Identity)-4.9/prebuilt/$(_PrebuiltToolchainDir)/bin/%(ToolPrefix)-%(Tool)"
```

Comment:
Should we make this line use `\`, since we mostly use that seperator in MSBuild targets?

I think `<Which/>` will return the OS-specific path.

Reply:
Yup, makes sense